static/js/util.ts

```diff
@@ -38,25 +43,28 @@ export function lower_bound(array, value, less) {
     return first;
 }
 
-export const lower_same = function lower_same(a, b) {
+export const lower_same = function lower_same(a: string, b: string): boolean {
     return a.toLowerCase() === b.toLowerCase();
 };
 
-export const same_stream_and_topic = function util_same_stream_and_topic(a, b) {
+export const same_stream_and_topic = function util_same_stream_and_topic(
+    a: Message,
+    b: Message,
```

Comment:
`same_stream_and_topic` is called with a non-`Message` here:

https://github.com/zulip/zulip/blob/f5f6a3789bafcffb7441fddc669a83b5aca7c76f/static/js/narrow.js#L317-L333